packages/binary/drng/subtypes/collectiveBeacon/payload/payload.go

```diff
@@ -0,0 +1,206 @@
+		return
+	}
+
+	// parse prevSignature
+	if result.prevSignature, err = marshalUtil.ReadBytes(SignatureSize); err != nil {
+		return
+	}
+
+	// parse current signature
+	if result.signature, err = marshalUtil.ReadBytes(SignatureSize); err != nil {
+		return
+	}
+
+	// parse distributed public key
+	if result.dpk, err = marshalUtil.ReadBytes(PublicKeySize); err != nil {
+		return
+	}
+
+	// return the number of bytes we processed
+	consumedBytes = marshalUtil.ReadOffset()
+
+	// store bytes, so we don't have to marshal manually
+	result.bytes = bytes[:consumedBytes]
+
+	return
+}
+
+func (payload *Payload) Bytes() (bytes []byte) {
+	// acquire lock for reading bytes
+	payload.bytesMutex.RLock()
+
+	// return if bytes have been determined already
+	if bytes = payload.bytes; bytes != nil {
+		defer payload.bytesMutex.RUnlock()
+		return
+	}
+
+	// switch to write lock
+	payload.bytesMutex.RUnlock()
+	payload.bytesMutex.Lock()
+	defer payload.bytesMutex.Unlock()
+
+	// return if bytes have been determined in the mean time
+	if bytes = payload.bytes; bytes != nil {
+		return
+	}
+
+	// marshal fields
+	payloadLength := header.Length + marshalutil.UINT64_SIZE + SignatureSize*2 + PublicKeySize
+	marshalUtil := marshalutil.New(marshalutil.UINT32_SIZE + marshalutil.UINT32_SIZE + payloadLength)
+	marshalUtil.WriteUint32(drngPayload.Type)
+	marshalUtil.WriteUint32(uint32(payloadLength))
+	marshalUtil.WriteBytes(payload.header.Bytes())
+	marshalUtil.WriteUint64(payload.Round())
+	marshalUtil.WriteBytes(payload.PrevSignature())
+	marshalUtil.WriteBytes(payload.Signature())
+	marshalUtil.WriteBytes(payload.DistributedPK())
+
+	bytes = marshalUtil.Bytes()
+
+	// store result
+	payload.bytes = bytes
+
+	return
+}
+
+func (payload *Payload) String() string {
+	return stringify.Struct("Payload",
+		stringify.StructField("type", uint64(payload.SubType())),
+		stringify.StructField("instance", uint64(payload.Instance())),
+		stringify.StructField("round", payload.Round()),
+		stringify.StructField("prevSignature", payload.PrevSignature()),
+		stringify.StructField("signature", payload.Signature()),
+		stringify.StructField("distributedPK", payload.DistributedPK()),
+	)
+}
+
+// region Payload implementation ///////////////////////////////////////////////////////////////////////////////////////
+
+func (payload *Payload) Type() payload.Type {
+	return drngPayload.Type
+}
+
+func (payload *Payload) Marshal() (bytes []byte, err error) {
+	return payload.Bytes(), nil
+}
+
+func (payload *Payload) Unmarshal(data []byte) (err error) {
+	_, err, _ = FromBytes(data, payload)
+
+	return
+}
+
+// func init() {
+// 	payload.RegisterType(drngPayload.Type, func(data []byte) (payload payload.Payload, err error) {
+// 		payload = &Payload{}
+// 		err = payload.UnmarshalBinary(data)
+
+// 		return
+// 	})
+// }
+
+// define contract (ensure that the struct fulfills the corresponding interface)
+var _ payload.Payload = &Payload{}
```

Comment:
Remove this please. The IDE should tell automatically whether `Payload` implements `payload.Payload`